@commitlint/types/src/load.ts

```diff
@@ -4,7 +4,7 @@ export type PluginRecords = Record<string, Plugin>;
 
 export interface Plugin {
 	rules: {
-		[ruleName: string]: Rule<unknown>;
+		[ruleName: string]: Rule;
```

Comment:
I've just ran into the same problem, you probably want to extend this rule to more than just the generic rule:
```suggestion
		[ruleName: string]: Rule | AsyncRule | SyncRule;
```

Reply:
hm, can't tag @jdbruijn here. @byCedric @iamscottcab what do you think?

Reply:
@chanceaclark Great catch, you're absolutely right! I've verified these locally by hanging what I assign to type `Plugin['rules']`. For full reference, I'm testing this in my plugin with the following, simplified, code and changing the 

## `Rule`

```ts
type FunctionRule = (
  parsed: Commit,
  when: RuleConfigCondition,
) => RuleOutcome | Promise<RuleOutcome>;

const functionRule: Rule<FunctionRule> = (
  parsed: Commit,
  when: RuleConfigCondition = 'always',
  value: FunctionRule | undefined,
) => {
  ...
};

const rules: Plugin['rules'] = {
  'function-rules/body-case': functionRule,
};
```

## `AsyncRule`

```ts
type FunctionRule = (
  parsed: Commit,
  when: RuleConfigCondition,
) => Promise<RuleOutcome>;

const functionRule: AsyncRule<FunctionRule> = (
  parsed: Commit,
  when: RuleConfigCondition = 'always',
  value: FunctionRule | undefined,
) => {
  ...
};

const rules: Plugin['rules'] = {
  'function-rules/body-case': functionRule,
};
```

## `SyncRule`

```ts
type FunctionRule = (
  parsed: Commit,
  when: RuleConfigCondition,
) => RuleOutcome;

const functionRule: SyncRule<FunctionRule> = (
  parsed: Commit,
  when: RuleConfigCondition = 'always',
  value: FunctionRule | undefined,
) => {
  ...
};

const rules: Plugin['rules'] = {
  'function-rules/body-case': functionRule,
};
```

Reply:
@escapedcat I've pushed the changes from review by @chanceaclark.